### PR TITLE
Added sample project for deploying Redaction Rulesets using Gradle.

### PR DESCRIPTION
**Pull request by millerbill3**
Added sample project for deploying Redaction Rulesets using Gradle.
Also included optional Gradle task for validating Redaction Rulesets anytime after deploying

### CODE REVIEW

File: examples/redaction-ruleset-project/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+This project shows an example of how MarkLogic 9 Redaction Rulesets can be loaded into a schemas
+database from src/main/ml-schemas (the default path - this can be overridden via 
+mlSchemasPath). 
+
+Note that in order for this to work, the content-database.json file must specify the schema
+database that it's associated with. And in most cases, you'll want your own schemas database - not the default Schemas one - so schemas-database.json can be used to create own with a name based on mlAppName.
+
+Within each folder containing one or more Redaction Rulesets, you must provide a **collecitons.properties** and **permissions.properties** file.
```
Comment (rjrudin) [inline examples/redaction-ruleset-project/README.md]: Rename to "collections"

File: examples/redaction-ruleset-project/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+These files contain the definitions for the applicable collections to be applied to the rulesets as well as the document permissions.
+
+***Note***: Rulesets must have a .json or .xml file extension.
+
+See [Specifying collections and permissions](https://github.com/marklogic-community/ml-javaclient-util/wiki/Loading-files#specifying-collections-and-permissions) for information on how to appliy the collections and permission when the rulesets are loaded
```
Comment (rjrudin) [inline examples/redaction-ruleset-project/README.md]: "appliy" to "apply"

File: examples/redaction-ruleset-project/build.gradle [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+/*
+ * ml-gradle adds an instance of com.marklogic.appdeployer.AppConfig to the Gradle project under the key "mlAppConfig".
+ * This instance can be modified to affect the behavior of ml-gradle.
+ */
+ext {
+	// mlAppConfig is an instance of com.marklogic.appdeployer.AppConfig
+	mlAppConfig {
+		// XCC URL for running corb task below and for creating triggers on pre-8.0-4 builds of MarkLogic
```
Comment (rjrudin) [inline examples/redaction-ruleset-project/build.gradle]: Change this comment to just "XCC URL for running corb task below"

File: examples/redaction-ruleset-project/build.gradle [hunk 4]
```diff
@@ -0,0 +1,60 @@
+buildscript {
+	repositories {
+		jcenter()
+		// Needed for corb dependency: XCC
+		maven { url "http://developer.marklogic.com/maven2/" }
+	}
+	dependencies {
+		//Needed for CorbTask to dynamicaly generate properties from CORB Options class
```
Comment (rjrudin) [inline examples/redaction-ruleset-project/build.gradle]: Shorten to "Needed to corb task below"

File: examples/redaction-ruleset-project/build.gradle [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	// This configuration captures the dependencies for running corb (Content Reprocessing in Bulk).
+	// This is only needed if you want to run corb via Gradle tasks.
+	// If you do, using com.marklogic.gradle.task.CorbTask is a useful starting point, as shown below.
+	corb
+}
+
+dependencies {
+	// required to run CoRB2
+	corb 'com.marklogic:marklogic-corb:2.3.2'
+	// optional
```
Comment (rjrudin) [inline examples/redaction-ruleset-project/build.gradle]: Remove these lines, they're out of scope

File: examples/redaction-ruleset-project/build.gradle [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+configurations {
+	// This configuration captures the dependencies for running corb (Content Reprocessing in Bulk).
+	// This is only needed if you want to run corb via Gradle tasks.
+	// If you do, using com.marklogic.gradle.task.CorbTask is a useful starting point, as shown below.
+	corb
+}
+
+dependencies {
+	// required to run CoRB2
```
Comment (rjrudin) [inline examples/redaction-ruleset-project/build.gradle]: Just say "Needed to run corb task below"

File: examples/redaction-ruleset-project/build.gradle [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+repositories {
+	jcenter()
+
+	// Needed for corb dependency: XCC
+	maven { url "http://developer.marklogic.com/maven2/" }
+}
+
+configurations {
+	// This configuration captures the dependencies for running corb (Content Reprocessing in Bulk).
```
Comment (rjrudin) [inline examples/redaction-ruleset-project/build.gradle]: Just say "Needed to run corb task below"

File: examples/redaction-ruleset-project/src/main/ml-schemas/redactionRules/collections.properties [hunk 8]
```diff
@@ -0,0 +1,5 @@
+#Add one or more collecitons to add the ruleset to (comma separated)
```
Comment (rjrudin) [inline examples/redaction-ruleset-project/src/main/ml-schemas/redactionRules/collections.properties]: "collecitons" to "collections"

File: examples/redaction-ruleset-project/src/main/ml-schemas/redactionRules/permissions.properties [hunk 9]
```diff
@@ -0,0 +1,5 @@
+# Add one or more document permissions to apply to the ruleset
```
Comment (rjrudin) [inline examples/redaction-ruleset-project/src/main/ml-schemas/redactionRules/permissions.properties]: I think it's unlikely that permissions will need to be different per ruleset, while it's very likely that collections will need to be different. I don't want to give the impression that this file is needed, as it's most likely not - so remove it from the example, just keep the collections example (which is a very good one).

File: examples/redaction-ruleset-project/build.gradle [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+		// XCC URL for running corb task below and for creating triggers on pre-8.0-4 builds of MarkLogic
+		contentXccUrl = "xcc://${mlUsername}:${mlPassword}@${mlHost}:${mlRestPort}"
+	}
+}
+
+/*
+ * This optional task demonstrates validating redaction rules via a locally run CORB process.
+ * To execute this task simply run it after you have successfully deployed the redaction rulesets (via mlAppDeploy or mlLoadSchemas).
+ */
+task validateRedactionRulesetsAdhoc(type: com.marklogic.gradle.task.CorbTask) {
```
Comment (rjrudin) [inline examples/redaction-ruleset-project/build.gradle]: This a good idea for validating the rulesets - does it print the results of each validation call to stdout? How does the user know what happened when this task is run?

**Comment by rjrudin** [comment]
Thanks @millerbill3 . I'm wondering if, instead of a corb example that just validates each ruleset, an MLCP example that uses redaction - i.e. https://docs.marklogic.com/guide/mlcp/export#id_21533 - would be better. 

You could use the sample data in the MLCP docs - https://docs.marklogic.com/guide/mlcp/export#id_34059 - and import those with Gradle, and then export them with Gradle (both of those steps using MlcpTask), with the export using the redaction option. That would give us a live example of the docs.

File: examples/redaction-ruleset-project/build.gradle (hunk 10 above)
Comment (millerbill3) [reply examples/redaction-ruleset-project/build.gradle]: With the current implementation, it simply spits out the normal  CORB processing lines and if it fails to validate, you'll see a typically error output to the console. It's not sexy by any means.

**Comment by millerbill3** [comment]
@rjrudin Would you propose a separate example project to demonstrate validating the redaction rulesets or just amending the current project by removing CORB and replacing it with MLCP? Would you prefer this change prior to merging the code?

**Comment by rjrudin** [comment]
Sorry for the late response - just amend this project to use mlcp instead of corb.

**Pull request closed without merging**